.rhdh/docs/openshift.md

```diff
@@ -0,0 +1,255 @@
+
+**Procedure**
+
+1. Create a project in the OpenShift, if not present.
+   For more information about creating a project in OpenShift, see the [Red Hat OpenShift documentation](https://docs.openshift.com/container-platform/4.14/applications/projects/working-with-projects.html#creating-a-project-using-the-web-console_projects).
+2. Switch to **Developer** mode on your Red Hat OpenShift web console.
+3. Click **+Add**.
+4. From the **Developer Catalog** panel, click **Operator Backed**.
+5. Search for _Backstage_ in the search bar and select the **Backstage** card.
+6. Click **Create**.
+7. Click **Create** and wait for the database and Red Hat Developer Hub to start.
+8. Click the **Open URL** option to start using the Red Hat Developer Hub platform.
+
+![RHDH from Operator](images/rhdh_from_operator.png)
+
+### Configurations for Operator-backed RHDH
+
+#### Adding a custom application configuration file to OpenShift
+
+To access the Red Hat Developer Hub, you must add a custom application configuration file to OpenShift.
+In OpenShift, you can use the following content as a base template to create a ConfigMap named `app-config-rhdh`:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+     app:
+       baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>
+       title: Red Hat Developer Hub
+     backend:
+       auth:
+         keys:
+           - secret: "${BACKEND_SECRET}"
+       baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>
+       cors:
+         origin: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>
+```
+
+<details>
+<summary>Example</summary>
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+    app:
+      title: Red Hat Developer Hub
+      baseUrl: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+    backend:
+      auth:
+        keys:
+          - secret: "${BACKEND_SECRET}"
+      baseUrl: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+      cors:
+        origin: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+```
+
+</details>
+
+**Notes**:
+- There is a mandatory Backend Auth Key for Backstage. This will reference an environment variable backed by an OpenShift Secret.
+- You need to know the external URL of your Backstage instance and set it in the `app.baseUrl`, `backend.baseUrl` and `backend.cors.origin` fields of the application configuration.
+  By default, this will be named as follows: `https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>`,
```

Comment:
Can we specify how to know these parameters, specifically OPENSHIFT_CLUSTER_ROUTE_HOST ?

Reply:
what about `oc cluster-info` ?

Reply:
I have nothing against 'oc cluster-info' or any other documented methods :)

Reply:
`oc cluster-info` seems to return the control plane URL, not the cluster domain URL.
For example, I had a sample Backstage Route exposed at https://backstage-bs1-my-ns.apps.586d5a08b54d03a8eeda.hypershift.aws-2.ci.openshift.org/, while `oc cluster-info` returned `https://a082bf2f1976b4869842662c61c71403-082d7fd8d45a63c6.elb.us-east-1.amazonaws.com:6443` as the control plane URL.

I found that `oc get ingresses.config/cluster -o jsonpath='{.spec.domain}'` returns the right information.
Fixed in [`2c6d22e` (#118)](https://github.com/janus-idp/operator/pull/118/commits/2c6d22e80c8bba0cbfc3c9e955bee21a533eb023)